worker/codalabworker/local_run/docker_image_manager.py

```diff
@@ -3,218 +3,194 @@
 import time
 import traceback
 import logging
-import sys
+
+import docker
 
 from codalabworker.formatting import size_str
-from codalabworker.docker_client import DockerException
-from codalabworker.fsm import BaseDependencyManager, DependencyStage, StateTransitioner
 from codalabworker.state_committer import JsonStateCommitter
 from codalabworker.worker_thread import ThreadDict
 
 logger = logging.getLogger(__name__)
 
-DockerImageState = namedtuple(
-    'DockerImageState', 'stage digest killed dependents last_used message'
-)
-
-
-class DockerImageManager(StateTransitioner, BaseDependencyManager):
-    def __init__(self, docker, commit_file, max_images_bytes, max_age_failed_seconds=60):
-        super(DockerImageManager, self).__init__()
-        self.add_transition(DependencyStage.DOWNLOADING, self._transition_from_DOWNLOADING)
-        self.add_terminal(DependencyStage.READY)
-        self.add_terminal(DependencyStage.FAILED)
-
-        self._state_committer = JsonStateCommitter(commit_file)
-        self._docker = docker
-        self._images = {}  # digest -> DockerImageState
-        # digest -> {'thread': Thread, 'success': bool}
-        self._downloading = ThreadDict(fields={'success': False})
-        self._max_images_bytes = max_images_bytes
-        self._max_age_failed_seconds = max_age_failed_seconds
+ImageAvailabilityState = namedtuple('ImageAvailabilityState', 'state info')
```

Comment:
TODO: Update locations that use DockerImageManager to work with this new representation